### PR TITLE
SIGNL4 Notification Plugin Update

### PR DESCRIPTION
**Pull request by rons4**
Thank you for your interest in contributing to Checkmk!
Consider looking into [Readme](https://github.com/Checkmk/checkmk#want-to-contribute) regarding process details.

## General information

This is an update of the SIGNL4 notification plugin for better support for service alerting and two-way status updates.
We are the provider of SIGNL4.

If you should have any questions please just let me know.

### CODE REVIEW

**Comment by rons4** [comment]
I have read the CLA Document and I hereby sign the CLA or my organization already has a signed CLA.

**Comment by rons4** [comment]
recheck

**Comment by rons4** [comment]
recheck

**Comment by rons4** [comment]
recheck

**Comment by rons4** [comment]
recheck

File: cmk/notification_plugins/signl4.py [hunk 1]
```diff
@@ -24,24 +26,46 @@ def _signl4_url() -> str:
 
 
 def _signl4_msg(context: dict[str, str]) -> dict[str, object]:
-    host_name = context["HOSTNAME"]
```
Comment (makanakoeln) [inline cmk/notification_plugins/signl4.py:27]: should be always present, no need for .get()

File: cmk/notification_plugins/signl4.py [hunk 2]
```diff
@@ -24,24 +26,46 @@ def _signl4_url() -> str:
...
-    notification_type = context["NOTIFICATIONTYPE"]
```
Comment (makanakoeln) [inline cmk/notification_plugins/signl4.py:28]: should be always present, no need for .get()

File: cmk/notification_plugins/signl4.py [hunk 3]
```diff
@@ -24,24 +26,46 @@ def _signl4_url() -> str:
...
+        else:
+            description += " (" + host_state + ")"
     # Remove placeholder "$SERVICEPROBLEMID$" if exists
     if service_problem_id.find("$") != -1:
         service_problem_id = ""
-
     # Check if this is a new problem or a recovery
     s4_status = "new" if notification_type != "RECOVERY" else "resolved"
 
+    # Base64 encode the SERVICEDESC for matching updates for service alerts
```
Comment (makanakoeln) [inline cmk/notification_plugins/signl4.py:55]: Is the servicedescription enough to identify the correct matching on signl4?
What about multiple hosts/sites with the same service description?

File: cmk/notification_plugins/signl4.py (hunk 1 above)
Comment (makanakoeln) [inline cmk/notification_plugins/signl4.py:27]: should be always present, no need for .get()

**Review by makanakoeln — CHANGES_REQUESTED** [review]

File: cmk/notification_plugins/signl4.py (hunk 2 above)
Comment (rons4) [reply cmk/notification_plugins/signl4.py:28]: OK, thanks.

File: cmk/notification_plugins/signl4.py (hunk 3 above)
Comment (rons4) [reply cmk/notification_plugins/signl4.py:55]: Thanks and good question. I was looking for something like a service ID. The service descriptions seemed to be the most unique choice. Do you have any other idea or best practice?

Comment (makanakoeln) [reply cmk/notification_plugins/signl4.py:55]: host+service+site / host+site should be unique, you can get the site with 

`context["OMD_SITE"]`

**Review by makanakoeln — CHANGES_REQUESTED** [review]

**Comment by rons4** [comment]
Sorry for the confusion. I checked the code again and we don't just use the service description for matching, we already use a combination of several values.

  "X-S4-ExternalID": "Checkmk: "
        + host_name
        + "-"
        + host_problem_id
        + "-"
        + service_problem_id
        + service_desc_id_part,

We use this for matching and for sending updates from our side (SIGNL4) back to Checkmk.

I hope this is fine then.

**Review by makanakoeln — APPROVED** [review]

**Comment by makanakoeln** [comment]
merged today. Thanks for contributing

**Pull request closed without merging**